### DISCUSSION TITLE
Dependency Dashboard - "WARN: Invalid tool constraint"

### DISCUSSION THREAD

**Discussion by MartinX3** [renovatebot/renovate#18818] [Request Help] [answered]
Is there a way to see why this warning happen and what doesn't work?

**Comment by viceice** [comment]
yes, check debug logs

**Reply by MartinX3** [reply]
A issue ticket generated by renovate has debug logs?
I only find the PR debug logs.

**Reply by MartinX3** [reply]
Ah, the issue ticket needs also a link to the debug log like the PR's.

```
WARN: Invalid tool constraint(branch="renovate/webpack-5.x")
{
  "toolName": "pnpm",
  "constraint": "latest"
}
```

I'll check if open issue tickets exist, since `"packageManager": "pnpm@latest"` in the package.json should be fine.
It installs the latest pnpm for me.

**Reply by MartinX3** [reply]
I think if renovate also updates this field, I can specify a version now to fix the warning.

**Comment by viceice** [comment, marked as answer]
- #3945

**Reply by MartinX3** [reply]
Ahhhhhh, thank you. I searched for `packageManager`
